### PR TITLE
Update remaining dialogs (EXPOSUREAPP-14000)

### PR DESCRIPTION
**Pull request by SamuraiKek**
This task updates the remaining dialogs in the app to the new dialog fragment approach.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/tracing/ui/settings/TracingSettingsFragment.kt [hunk 1]
```diff
@@ -110,19 +110,10 @@ class TracingSettingsFragment : Fragment(R.layout.fragment_tracing_settings), Au
...
-            false,
-            {
-                // close dialog
-            }
-        )
-        DialogHelper.showDialog(dialog)
+    private fun showManualCheckingRequiredDialog() = displayDialog {
+        setTitle(R.string.onboarding_manual_required_dialog_headline)
+        setMessage(R.string.onboarding_manual_required_dialog_body)
+        setPositiveButton(R.string.onboarding_manual_required_dialog_button) { _, _ -> }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/tracing/ui/settings/TracingSettingsFragment.kt:116]: `displayDialog (cancelable=false)`

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragment.kt [hunk 2]
```diff
@@ -49,11 +49,11 @@ class DebugOptionsFragment : Fragment(R.layout.fragment_test_debugoptions), Auto
...
                 vm.selectEnvironmentType(type)
-                MaterialAlertDialogBuilder(requireContext())
-                    .setTitle("Restarting ↻")
-                    .setMessage("Configuring $type environment. Get yourself a glass of water \uD83D\uDEB0")
-                    .setCancelable(false)
-                    .show()
+                displayDialog {
+                    setTitle("Restarting ↻")
+                    setMessage("Configuring $type environment. Get yourself a glass of water \uD83D\uDEB0")
+                    setCancelable(false)
```
Comment (mtwalli) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragment.kt:55]: does `setCancelable`  here works ? should not you pass it in  displayDialog (cancelable=false)?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt [hunk 3]
```diff
@@ -1,141 +0,0 @@
...
-    private fun getMessage(context: Context, message: String?, isTextSelectable: Boolean): TextView {
-        // create spannable and add links, removed stack trace links into nowhere
-        val spannable = SpannableString(message)
-        val httpPattern: Pattern = Pattern.compile("[a-z]+://[^ \\n]*")
-        Linkify.addLinks(spannable, httpPattern, "")
-        // get padding for all sides
-        val paddingStartEnd = context.resources.getDimension(R.dimen.spacing_normal).toInt()
-        val paddingLeftRight = context.resources.getDimension(R.dimen.spacing_small).toInt()
-        // create a textview with clickable links from the spannable
-        val textView = TextView(context)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt:131]: I think some error dialogs have links, will they be still working ?

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragment.kt (hunk 2 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragment.kt:55]: Yep, you're right. I just copy pasted what was inside the Builder and didn't notice it.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt (hunk 3 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt:131]: I'll check.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/dialog/DccTicketingDialogs.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.ui.dialog.displayDialog
+
+fun Fragment.dccTicketingConfirmCancellationDialog(negativeButtonAction: () -> Unit) = displayDialog {
+    setTitle(R.string.dcc_ticketing_consent_one_cancel_dialog_title)
+    setMessage(R.string.dcc_ticketing_consent_one_cancel_dialog_body)
+    setPositiveButton(R.string.dcc_ticketing_consent_one_cancel_dialog_continue_btn) { _, _ -> }
+    setNegativeButton(R.string.dcc_ticketing_consent_one_cancel_dialog_cancel_btn) { _, _ -> }
```
[the remark this answers is not in the document]
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/dialog/DccTicketingDialogs.kt]: Yes. I've already updated it. I don't know why this still shows up with the old `displayDialog()` function

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/exception/reporting/ErrorReportReceiver.kt [hunk 5]
```diff
@@ -62,26 +63,17 @@ class ErrorReportReceiver(private val activity: Activity) : BroadcastReceiver()
...
-            negativeButtonFunction = {
-                val stackTraceDialog = DialogHelper.DialogInstance(
-                    activity,
-                    title,
-                    "$detailsTitle:\n$stack",
-                    confirm
-                )
-                DialogHelper.showDialog(stackTraceDialog.copy(isTextSelectable = true))
-                Unit
+        (activity as AppCompatActivity).createDialog {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/exception/reporting/ErrorReportReceiver.kt:66]: what we need to from the activity is just the context, can we make the extension function for Activity directly 
` fun Activity.createDialog()`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+data class CwaDialogTexts(
+    val title: @RawValue IntOrString? = null,
+    val message: @RawValue IntOrString? = null,
+    val positiveButtonText: @RawValue IntOrString? = null,
+    val negativeButtonText: @RawValue IntOrString? = null,
+    val neutralButtonText: @RawValue IntOrString? = null,
+) : Parcelable
+
+@CwaDialogDsl
+class CwaDialogBuilder {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt:24]: maybe in another PR , could you write some unit tests for this class ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialogHelper.kt [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+ *     negativeButton("Negative button text") { foo2() }
+ *     neutralButton("Neutral button text") { foo3() }
+ *     dismissAction { foo4() }
+ *     setCancelable(false)
+ *     setDeleteDialog(true)
+ *     setCustomView(R.layout.dialog_view)
+ * }
+ * ```
+ */
+fun Fragment.createDialog(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialogHelper.kt:24]: should we keep the old name `display*` it is not just creating the dialog, it does show it too ?

Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialogHelper.kt:24]: Yes, you're right. I had them in parallel and had to name this one something different.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt (hunk 6 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt:24]: Alright.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/exception/reporting/ErrorReportReceiver.kt (hunk 5 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/exception/reporting/ErrorReportReceiver.kt:66]: I tried doing this but I need a fragment manager and I'm not sure how to obtain one for the `Activity`. It says it's deppracated.

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/exception/reporting/ErrorReportReceiver.kt:66]: Ah, ok

**Review by mtwalli — APPROVED** [review]

**Pull request merged**